Job Scheduler/JobScheduler.java

```diff
@@ -0,0 +1,198 @@
+	 * @param localarr localarr depicts the sorted array which is to be worked upon
+	 * @param completionarr depicts the array with the completion time of processes 
+	 * @return an array with the complete time of the processes
+	 */
+	
+	int[] calculateCompletionTime(int localarr[][],int completionarr[])
+	{
+	    completionarr[0]=localarr[0][0]+localarr[0][1];
+	    for(int i=1;i<localarr.length;i++)
+	    {
+	        if(completionarr[i-1]>localarr[i][0])
+	            completionarr[i]=completionarr[i-1]+localarr[i][1];
+	        else
+	            completionarr[i]=localarr[i][0]+localarr[i][1];
+	    }
+	    return completionarr;
+	}
+	
+	/**
+	 * 
+	 * @param localarr depicts the sorted array which is to be worked upon
+	 * @param turnaroundarr depicts an array of Time taken to complete after arrival
+	 * @param completionarr depicts the array with the completion time of processes
+	 * @return an array with the turnaround time for processes
+	 */
+	int[] calculateTurnAroundTime(int [][]localarr,int turnaroundarr[],int completionarr[])
+	{
+	    for(int i=0;i<localarr.length;i++)
+	    {
+	        turnaroundarr[i]=completionarr[i]-localarr[i][0];
+	    }
+	    return turnaroundarr;
+	}
+	
+	/**
+	 * 
+	 * @param localarr  depicts the sorted array which is to be worked upon
+	 * @param waitingarr depicts an array of Total time the process has to wait before it's execution begins
+	 * @param turnaroundarr depicts an array of Time taken to complete after arrival
+	 * @return the array with waiting time of the processes
+	 */
+	int[] calculateWaitingTime(int localarr[][],int waitingarr[],int turnaroundarr[])
+	{
+	    for(int i=0;i<localarr.length;i++)
+	    {
+	        waitingarr[i]=turnaroundarr[i]-localarr[i][1];
+	    }
+	    return waitingarr;
+	}
+	
+	/**
+	 * Calculates the average waiting time of the processes
+	 * @param waitingarr depicts an array of Total time the process has to wait before it's execution begins
+	 * @return the average waiting time of the processes
+	 */
+	double averageWaitingTime(int waitingarr[])
+	{
+	    int totalWaitingtime=waitingarr[0];
+	    double averageWaitingtime=0.0;
+	    for(int i=1;i<waitingarr.length;i++)
+	    {
+	        totalWaitingtime+=waitingarr[i];
+	    }
+	    averageWaitingtime=(double)totalWaitingtime/(waitingarr.length);
+	    return averageWaitingtime ;
+	}
+	
+	/**
+	 * Calculates the maximum waiting time among the waiting time of all the processes
+	 * @param waitingarr depicts an array of Total time the process has to wait before it's execution begins
+	 * @return the maximum waiting time of the processes
+	 */
+	int maxWaitingTime(int waitingarr[])
+	{
+	    int maxwaitingtime=waitingarr[0];
+	    for(int i=1;i<waitingarr.length;i++)
+	    {
+	        if(maxwaitingtime<waitingarr[i])
+	            maxwaitingtime=waitingarr[i];
+	    }
+	    return maxwaitingtime;
+	}
+	
+	public static void main(String[] args) 
+	{
+		JobScheduler obj1=new JobScheduler();

```

Comment:
Rename obj1 to name line jobSchedulerInstance or any thing which shows its purpose